### DISCUSSION TITLE
How to interface with the JVM?

### DISCUSSION THREAD

**Discussion by ysthakur** [factor/factor#2453] [Q&A]
Is it possible to call Java methods from Factor and vice versa? It seems Factor used to run on the JVM but not anymore, so I can't tell if I can call a Factor program from a JVM language.

**Comment by catb0t** [comment]
lazy answer: if you have a C FFI for your Java library, then Factor can interface with that, but that's 2 levels of type conversions

**Reply by ysthakur** [reply]
Thanks for your answer! I guess that might work, but at that point I may as well just use a JVM language for everything or rewrite the library I need (asm) for Factor, although the second is not a viable option either.

**Comment by mrjbq7** [comment]
What library is that?

**Reply by ysthakur** [reply]
[ASM](https://asm.ow2.io/), for reading and manipulating Java bytecode.
